Scripts/prepare_release.py

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/env python
```

Comment:
I have a challenging question for you. How hard would it be to actually convert this code to dafny so that it can be run with `dafny -noVerify ` ?
That way, you wouldn't need to have `assert ends_with('...')` that I see in the code.
Another advantage would be that we would be able to execute tests that would greatly enhance the maintenance process. There are tons of regexps that might break that would be easy to fix if only we saw what the typical input and typical output should look like.

Reply:
It would be anywhere between a huge pain and impossible, because of the range of libraries used: XML parsing, regular expressions, HTTP requests, dates and times, text manipulation, file system functions, quoting, etc (but not JSON parsing, because we do have that! ^^)

I can remove these assertions if you'd like; they'd just become requires in Dafny.  They are here to make sure that the messages are readable.

Python has better support for tests than Dafny; I can add tests if you want (but there's decreasing returns to spending days working on a script that runs once a month ^^)

Reply:
We will probably want to run this script once a week. I'm not saying we have to convert this script to Dafny, but if it's challenging to write such a script in Dafny, then maybe it's a yellow flag as per how difficult it is to integrate with an existing environment.
Dafny compiles to python as well, so it should not be hard to have wrapper libraries.

Reply:
It's hundreds of lines of wrapper code / dozens of wrappers; I don't think Dafny is ready to be a scripting language (and that's OK)